### PR TITLE
V0.3beta

### PR DESCRIPTION
**Pull request by lberrada**
* New metrics (BestMetric, DynamicMetric)
* Logging out stdout
* New classes for cleaner code (Plotter, Index)
* Simpler syntax to access and log metrics
* Documentation in README

### CODE REVIEW

File: logger/plotter.py [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+        Returns True / non-empty string if data has been sent successfully,
+        False otherwise.
+        """
+        xlabel = 'Time (s)' if time_idx else 'Index'
+        if name not in list(self.windows.keys()):
+            opts = {'legend': [tag], 'title': name, 'xlabel': xlabel}
+            self.windows[name] = self.viz.line(Y=y, X=x, opts=opts)
+            return True
+        else:
+            return self.viz.updateTrace(Y=y, X=x, name=tag,
```
Comment (albanD) [inline logger/plotter.py]: So this returns False if it fails? Not hang?

File: logger/plotter.py [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+        self.viz = visdom.Visdom(**visdom_opts)
+        self.windows = {}
+        self.append = {}
+        self.cache = defaultdict(Cache)
+
+    def _plot_xy(self, name, tag, x, y, time_idx=True):
+        """
+        Creates a window if it does not exist yet.
+        Returns True / non-empty string if data has been sent successfully,
```
Comment (albanD) [inline logger/plotter.py]: Maybe add an extra check so that the function returns just True/False ?

File: logger/xp.py [hunk 3]
```diff
@@ -1,34 +1,27 @@
...
 
 class Experiment(object):
 
     def __init__(self, name, log_git_hash=True,
-                 use_visdom=False, visdom_opts={}):
+                 use_visdom=False, visdom_opts=None,
+                 time_indexing=True):
         """ Create an experiment with the following parameters:
         - log_git_hash (bool): retrieve current commit hash to log code status
         - use_visdom (bool): monitor metrics logged on visdom
```
Comment (albanD) [inline logger/xp.py]: Missing doc for `time_indexing`.

File: logger/plotter.py (hunk 1 above)
Comment (lberrada) [reply logger/plotter.py]: Actually yes, the error is caught [there](https://github.com/facebookresearch/visdom/blob/master/py/__init__.py#L266)

**Pull request merged**